packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -288,7 +288,11 @@ export interface SdkCredentialType extends SdkTypeBase {
 export interface SdkModelPropertyTypeBase {
   __raw?: ModelProperty;
   type: SdkType;
+  /**
```

Comment:
Can we create a follow up to remove the `nameInClient` method and link the issue here so we don't forget.

Given tcgc is all internal we can be more aggressive with breaking changes if we all agree.